x-pack/plugins/apm/dev_docs/testing.md

```diff
@@ -154,3 +156,31 @@ All files with a .stories.tsx extension will be loaded. You can access the devel
 For end-to-end (e.g. agent -> apm server -> elasticsearch <- kibana) development and testing of Elastic APM please check the the [APM Integration Testing repository](https://github.com/elastic/apm-integration-testing).
 
 Data can also be generated using the [kbn-apm-synthtrace](../../../../packages/kbn-apm-synthtrace/README.md) CLI.
+
+## Best practices for API tests
```

Comment:
We have a lot of unit tests that are mocking out Elasticsearch and should probably have been API tests in the first place. Example: https://github.com/elastic/kibana/blob/07f1c36df72a61e6a86d617e5170d3d1f3b2b504/x-pack/plugins/apm/server/routes/errors/distribution/queries.test.ts. We have [an issue](https://github.com/elastic/kibana/issues/152557) that captures that problem.

Perhaps add a comment when something should be an API test instead of a unit test.

Reply:
Added a comment in [`082ca04` (#167507)](https://github.com/elastic/kibana/pull/167507/commits/082ca04138f461075bb24975c1ac495f82c02949)